File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
arr = [list(input()) for _ in range(N)]
result = N * M + 1

for i in range(N-7):
    for j in range(M-7):
        w_cnt = 0
        b_cnt = 0

        for a in range(i, i+8): # i부터 8칸까지
            for b in range(j, j+8):  # j부터 8칸까지
                if (a + b) % 2:      # a+b가 홀수이면
                    if arr[a][b] != 'B':
                        w_cnt += 1
                    else:
                        b_cnt += 1

                else:
                    if arr[a][b] != 'W':
                        w_cnt += 1
                    else:
                        b_cnt += 1
        result = min(w_cnt, b_cnt, result)
print(result)

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static int result = 64;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        // W면 0, B면 1
        int[][] arr = new int[n][m];

        for(int i = 0; i < n; i++) {
            String str = br.readLine();

            for (int j = 0; j < m; j++) {
                if(str.charAt(j) == 'W') arr[i][j] = 0;
                else arr[i][j] = 1;
            }
        }

        for (int i = 0; i < n - 7; i++) {
            for (int j = 0; j < m - 7; j++) {
                check(i, j, arr);
            }
        }
        System.out.println(result);
    }

    public static void check(int x, int y, int[][] arr) {
        int cnt = 0;
        int color = arr[x][y];  // 첫번째 칸 색 (W: 0, B: 1)

        for (int i = x; i < x + 8; i++) {
            for (int j = y; j < y + 8; j++) {
                if(arr[i][j] != color) cnt++;

                color = (color + 1) % 2;    // 다음 칸은 반대 색
            }
            color = (color + 1) % 2;
        }
        cnt = Math.min(cnt, 64 - cnt);
        result = Math.min(result, cnt);
    }
}